frontend/src/components/AdminUser/AdminUserBody.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { User } from '.';
+import {
+  TableBodyRow,
+  TableBodyRowData,
+  TableBodyRowAvatar,
+  TableBodyRowStatus,
+  TableBody,
+} from '../Table';
+import { AdminUserAction } from './AdminUserAction';
+
+interface AdminUserBodyProps {
+  users?: User[];
+}
+
+export const AdminUserBody = ({ users }: AdminUserBodyProps): JSX.Element => {
+  function renderUsers() {
+    return users?.map((user) => {
+      return (
+        <TableBodyRow key={user.id}>
+          <TableBodyRowData>
+            <div className="flex items-center">
+              <TableBodyRowAvatar src={user.avatar} />
+              <div className="ml-4">{user.name}</div>
+            </div>
+          </TableBodyRowData>
+          <TableBodyRowData>{user.email}</TableBodyRowData>
+          <TableBodyRowData>
+            <TableBodyRowStatus
+              text={user.is_admin ? 'Admin' : 'User'}
+              statusBg={user.is_admin ? 'green-100' : 'cyan-100'}
+              statusText={user.is_admin ? 'green-900' : 'cyan-900'}
+            />
+          </TableBodyRowData>
+          <AdminUserAction user={user} />
+        </TableBodyRow>
+      );
+    });
+  }
+
+  return <>{renderUsers()}</>;
+};
```

Comment:
I just realize this is a bit unnecessarily decoupled. Unless you have plans to reuse renderUsers() function i think you wouldnt need to extract it to a function pa you simply have it like so:
```js
export const AdminUserBody = ({ users }: AdminUserBodyProps): JSX.Element => {

  return (
    <>
      { users?.map((user) => {
        return (
          <TableBodyRow key={user.id}>
            <TableBodyRowData>
              <div className="flex items-center">
                <TableBodyRowAvatar src={user.avatar} />
                <div className="ml-4">{user.name}</div>
              </div>
            </TableBodyRowData>
            <TableBodyRowData>{user.email}</TableBodyRowData>
            <TableBodyRowData>
              <TableBodyRowStatus
                text={user.is_admin ? 'Admin' : 'User'}
                statusBg={user.is_admin ? 'green-100' : 'cyan-100'}
                statusText={user.is_admin ? 'green-900' : 'cyan-900'}
              />
            </TableBodyRowData>
            <AdminUserAction user={user} />
          </TableBodyRow>
        );
      })}
    </>
  );
};
```

Reply:
changed to suggestion